source/Cosmos.System2/Graphics/Bitmap.cs

```diff
@@ -0,0 +1,226 @@
+            uint offset = 54;
+            data = BitConverter.GetBytes(offset);
+            Array.Copy(data, 0, file, position, 4);
+            position += 4;
+
+            //Write size of bitmapinfoheader
+            data = BitConverter.GetBytes(40);
+            Array.Copy(data, 0, file, position, 4);
+            position += 4;
+
+            //Width in pixels
+            data = BitConverter.GetBytes(width);
+            Array.Copy(data, 0, file, position, 4);
+            position += 4;
+
+            //Height in pixels
+            data = BitConverter.GetBytes(height);
+            Array.Copy(data, 0, file, position, 4);
+            position += 4;
+
+            //Number of palnes(1)
+            data = BitConverter.GetBytes(1);
+            Array.Copy(data, 0, file, position, 2);
+            position += 2;
+
+            //Bits per pixel
+            data = BitConverter.GetBytes((int)depth);
+            Array.Copy(data, 0, file, position, 2);
+            position += 2;
+
+            //Compression type
+            data = BitConverter.GetBytes(0);
+            Array.Copy(data, 0, file, position, 4);
+            position += 4;
+
+            //Size of image data in bytes
+            data = BitConverter.GetBytes(width * height * (uint)depth / 8);
+            Array.Copy(data, 0, file, position, 4);
+            position += 4;
+
+            //Horizontal resolution in meters (is not accurate)
+            data = BitConverter.GetBytes(width / 40);
+            Array.Copy(data, 0, file, position, 4);
+            position += 4;
+
+            //Vertical resolution in meters (is not accurate)
+            data = BitConverter.GetBytes(height / 40);
+            Array.Copy(data, 0, file, position, 0);
+            position += 4;
+
+            //Number of colors in image /zero
+            data = BitConverter.GetBytes(0);
+            Array.Copy(data, 0, file, position, 0);
+            position += 4;
+
+            //number of important colors in iamge / zero
+            data = BitConverter.GetBytes(0);
+            Array.Copy(data, 0, file, position, 4);
+            position += 4;
+
+            //Finished header
+
+            //Copy image data
+            position = (int)offset;
+            Byte[] imageData = new Byte[width * height * (int)depth / 8];
+            int imageDataPoint = 0;
+            for (int x = 0; x < width; x++)
+            {
+                for (int y = 0; y < height; y++)
+                {
+                    data = BitConverter.GetBytes(rawData[x + (height - (y + 1)) * width]);
+                    Array.Copy(data, 0, imageData, imageDataPoint, data.Length);
+                    imageDataPoint += data.Length;
+                }
+            }
+            Array.Copy(imageData, 0, file, position, imageData.Length);
+            stream.Write(file, 0, file.Length);
+        }
+
+        public static Bitmap CreateBitmap(string path)
+        {
+            #region BMP Header
+
+            Bitmap bitmap;
```

Comment:
I would prefer to have ctor instead of a static method to create a bitmap, we could have:

```cs
Bitmap(String); // Create a Bitmap from a file
Bitmap(byte[]); // Create a Bitmap from a byte array 
Bitmap(Stream); // Create a Bitmap from a Stream
```

the first one could be simply empty doing:
```cs
Bitmap(String path) {
    using (FileStream fs = new FileStream(path, FileMode.Open)) {
           return new Bitmap(fs);
    }
```
Bitmap(byte[]) is similar it creates a Stream using MemoryStream(array), only the last one does really something and it pratically the code you have written between "using" in CreateBitmap().